src/en/anchira/src/eu/kanade/tachiyomi/extension/en/anchira/Anchira.kt

```diff
@@ -0,0 +1,342 @@
+class Anchira : HttpSource(), ConfigurableSource {
+    override val name = "Anchira"
+
+    override val baseUrl = "https://anchira.to"
+
+    private val apiUrl = "$baseUrl/api/v1"
+
+    private val libraryUrl = "$apiUrl/library"
+
+    private val cdnUrl = "https://kisakisexo.xyz"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3, 1, TimeUnit.SECONDS)
+        .addInterceptor { apiInterceptor(it) }
+        .build()
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override fun headersBuilder() = super.headersBuilder().add("X-Requested-With", "XMLHttpRequest")
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int) = GET("$libraryUrl?page=$page", headers)
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val data = decodeBytes<LibraryResponse>(response.body)
+
+        return if (data.entries != null) {
+            MangasPage(
+                data.entries.map {
+                    SManga.create().apply {
+                        url = "/g/${it.id}/${it.key}"
+                        title = it.title
+                        thumbnail_url = "$cdnUrl/${it.id}/${it.key}/m/${it.cover.name}"
+                        artist = it.tags.filter { it.namespace == 1 }.joinToString(", ") { it.name }
+                        author = it.tags.filter { it.namespace == 2 }.joinToString(", ") { it.name }
+                        genre = prepareTags(it.tags)
+                        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+                        status = SManga.COMPLETED
+                        initialized = false
+                    }
+                }.toList(),
+                data.entries.size < data.total,
+            )
+        } else {
+            MangasPage(listOf(), false)
+        }
+    }
+
+    // Popular
+
+    override fun popularMangaRequest(page: Int) = GET("$libraryUrl?sort=32&page=$page", headers)
+
+    override fun popularMangaParse(response: Response) = latestUpdatesParse(response)
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        var url = libraryUrl.toHttpUrl().newBuilder()
+
+        url.addQueryParameter("page", page.toString())
+
+        if (query.isNotBlank()) {
+            url.addQueryParameter("s", query)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is CategoryGroup -> {
+                    var sum = 0
+
+                    filter.state.forEach { category ->
+                        when (category.name) {
+                            "Manga" -> if (category.state) sum += 1
+                            "Doujinshi" -> if (category.state) sum += 2
+                            "Illustration" -> if (category.state) sum += 4
```

Comment:
```suggestion
                            "Manga" -> if (category.state) sum = sum or 1
                            "Doujinshi" -> if (category.state) sum = sum or 2
                            "Illustration" -> if (category.state) sum = sum or 4
```

Maybe use bitwise OR for these instead. Adding also works, but it is better to use bitwise OR for bit flags.